### ISSUE TITLE
Numbers comparison issue when json path is used

### ISSUE THREAD

**Issue by orange-buffalo** [lukas-krecan/JsonUnit#144]
Please consider the following code:
```java
assertThatJson("{\"root\": [{\"target\": 450} ]}")
                .inPath("$.root")
                .isArray()
                .containsExactly("{target: 450 }");
```
My expectation is that this assertion is satisfied, it looks fully valid to me.
But it fails with the following comparison issue:
```
2018-10-29 09:37:51.894 DEBUG   --- [           main] net.javacrumbs.jsonunit.difference.diff  : JSON documents are different:
Different value found in node "$.root", expected: <{"target":4.5E+2}> but was: <"{target: 450 }">.
2018-10-29 09:37:51.904 DEBUG   --- [           main] net.javacrumbs.jsonunit.difference.diff  : JSON documents are different:
Different value found in node "$.root.target", expected: <450> but was: <4.5E+2>.

java.lang.AssertionError: [Different value found in node "$.root"] 
Expecting:
  <[{"target":450}]>
to contain exactly (and in same order):
  <["{target: 450 }"]>
but some elements were not found:
  <["{target: 450 }"]>
and others were not expected:
  <[{"target":450}]>
when comparing values using JsonComparator
```
(a unit test to reproduce this is [here](https://github.com/orange-buffalo/JsonUnit/blob/numbers-in-path/tests/test-json-path/src/test/java/net/javacrumbs/jsonunit/test/jsonpath/JsonPathNumbersTest.java)).

This is not reproduced when comparing objects without json path. I dare to suggest that json path compacts the big decimals, because the actual and expected node values have different `scale` value. Consequently, [Diff](https://github.com/lukas-krecan/JsonUnit/blob/master/json-unit-core/src/main/java/net/javacrumbs/jsonunit/core/internal/Diff.java#L307) fails, because it compares values by `equals` (two big decimals with different scales are not equal). 

I believe that in general, it is better to compare `BigDecimal` values using `compareTo` method. It will handle any differences in scaling correctly. Would you accept a PR that extends `net.javacrumbs.jsonunit.core.internal.Diff#compareValues` to check if values are `Comparable` and uses `compareTo` in this case?

**Comment by lukas-krecan** [comment]
Hi, thanks for feedback. It looks similar to https://github.com/lukas-krecan/JsonUnit/issues/130 will take a look at it, in the meantime you can use `.withConfiguration(c -> c.withTolerance(0))`

**Comment by orange-buffalo** [comment]
Thanks for the workaround! And sorry for the duplicate, didn't find 130 when searched for the similar issues.

**Comment by lukas-krecan** [comment]
It's not a duplicate, it's a brand new bug. Just released 2.0.2 with a fix. Please let me know whether it works

**Comment by orange-buffalo** [comment]
I've tested with the new version and it works as expected. Many thanks for the quick resolution!
